### PR TITLE
Moves to the template loader hooks and functions into lib/compat folder

### PR DESCRIPTION
**Pull request by youknowriad**
See #37141 for the reasoning here.

### CODE REVIEW

File: lib/compat/wordpress-5.9/block-template.php [hunk 1]
```diff
@@ -223,13 +224,6 @@ function gutenberg_get_the_template_html() {
 	return '<div class="wp-site-blocks">' . $content . '</div>';
 }
 
-/**
- * Renders the markup for the current template.
- */
-function gutenberg_render_the_template() {
```
Comment (youknowriad) [inline lib/compat/wordpress-5.9/block-template.php:229]: I believe that function is useless and I can't find it on Core either. So either we remove it or backport it.

Comment (youknowriad) [reply lib/compat/wordpress-5.9/block-template.php:229]: @scruffian Is this something that is needed for universal themes? Should this be backported as `render_block_template` or something in Core?

File: lib/compat/wordpress-5.9/block-template.php [hunk 2]
```diff
@@ -17,7 +17,7 @@ function gutenberg_add_template_loader_filters() {
 		if ( 'embed' === $template_type ) { // Skip 'embed' for now because it is not a regular template type.
 			continue;
 		}
-		add_filter( str_replace( '-', '', $template_type ) . '_template', 'gutenberg_override_query_template', 20, 3 );
+		add_filter( str_replace( '-', '', $template_type ) . '_template', 'gutenberg_locate_block_template', 20, 3 );
```
Comment (youknowriad) [inline lib/compat/wordpress-5.9/block-template.php:20]: just aligning function names with Core equivalents

File: lib/compat/wordpress-5.9/block-template.php (hunk 1 above)
Comment (ockham) [reply lib/compat/wordpress-5.9/block-template.php:229]: Seems to me like this has been obsolete since #28319 🤔

**Review by ockham — CHANGES_REQUESTED** [review]
Since we're moving `template-canvas.php`, I believe we also need to update this path:

```diff
diff --git a/lib/compat/wordpress-5.9/block-template.php b/lib/compat/wordpress-5.9/block-template.php
index 85b0a91b4e..4a57ece0d8 100644
--- a/lib/compat/wordpress-5.9/block-template.php
+++ b/lib/compat/wordpress-5.9/block-template.php
@@ -102,7 +102,7 @@ function gutenberg_locate_block_template( $template, $type, array $templates ) {
        add_action( 'wp_head', 'gutenberg_render_title_tag', 1 ); // ...and make it unconditional.
 
        // This file will be included instead of the theme's template file.
-       return gutenberg_dir_path() . 'lib/template-canvas.php';
+       return gutenberg_dir_path() . 'lib/compat/wordpress-5.9/template-canvas.php';
 }
 
 /**
```

**Comment by youknowriad** [comment]
@ockham good catch, I actually did that change but somehow it got lost before the commit.

**Review by ockham — APPROVED** [review]
Thanks Riad, this LGTM now! 👍 

(We might want to wait for confirmation for [this question](https://github.com/WordPress/gutenberg/pull/37149/files#r762992787) before merging.)

File: lib/compat/wordpress-5.9/block-template.php (hunk 1 above)
Comment (youknowriad) [reply lib/compat/wordpress-5.9/block-template.php:229]: Indeed 👍 I was confusing this with `block_template_part` which is used in hybrid themes.

**Pull request merged**